// aurelia-validation/dist/test/test/resources/number-value.js
define(["require", "exports", "tslib", "aurelia-pal", "aurelia-dependency-injection", "aurelia-templating", "aurelia-binding"], function (require, exports, tslib_1, aurelia_pal_1, aurelia_dependency_injection_1, aurelia_templating_1, aurelia_binding_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var NumberBase = /** @class */ (function () {
        function NumberBase(input) {
            var _this = this;
            this.input = input;
            this.inputValueChanged = function () {
                _this.value = _this.input.value === '' ? null : parseInt(_this.input.value, 10);
            };
        }
        NumberBase.prototype.valueChanged = function (newValue) {
            this.input.value = newValue === null ? '' : newValue.toString(10);
        };
        NumberBase.prototype.bind = function () {
            this._input = this.input;
            this._input.addEventListener('change', this.inputValueChanged);
        };
        NumberBase.prototype.unbind = function () {
            this._input.removeEventListener('change', this.inputValueChanged);
            this._input = null;
        };
        return NumberBase;
    }());
    exports.NumberBase = NumberBase;
    var NumberValueCustomAttribute = /** @class */ (function (_super) {
        tslib_1.__extends(NumberValueCustomAttribute, _super);
        function NumberValueCustomAttribute() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        NumberValueCustomAttribute = tslib_1.__decorate([
            aurelia_templating_1.customAttribute('number-value', aurelia_binding_1.bindingMode.twoWay),
            aurelia_dependency_injection_1.inject(Element)
        ], NumberValueCustomAttribute);
        return NumberValueCustomAttribute;
    }(NumberBase));
    exports.NumberValueCustomAttribute = NumberValueCustomAttribute;
    var NumberInputCustomElement = /** @class */ (function (_super) {
        tslib_1.__extends(NumberInputCustomElement, _super);
        function NumberInputCustomElement(element) {
            var _this = _super.call(this, null) || this;
            _this.element = element;
            _this.inputBlurred = function () {
                _this.element.dispatchEvent(aurelia_pal_1.DOM.createCustomEvent('blur', {}));
            };
            _this.element.focus = function () { return _this.input.focus(); };
            return _this;
        }
        NumberInputCustomElement.prototype.bind = function () {
            _super.prototype.bind.call(this);
            this._input.addEventListener('blur', this.inputBlurred);
        };
        NumberInputCustomElement.prototype.unbind = function () {
            this._input.removeEventListener('blur', this.inputBlurred);
            _super.prototype.unbind.call(this);
        };
        tslib_1.__decorate([
            aurelia_templating_1.bindable({ defaultBindingMode: aurelia_binding_1.bindingMode.twoWay })
        ], NumberInputCustomElement.prototype, "value", void 0);
        NumberInputCustomElement = tslib_1.__decorate([
            aurelia_templating_1.customElement('number-input'),
            aurelia_dependency_injection_1.inject(Element),
            aurelia_templating_1.inlineView("<template><input ref=\"input\"></template>")
        ], NumberInputCustomElement);
        return NumberInputCustomElement;
    }(NumberBase));
    exports.NumberInputCustomElement = NumberInputCustomElement;
});
