// aurelia-validation/dist/test/test/resources/trigger-form.js
define(["require", "exports", "tslib", "aurelia-dependency-injection", "aurelia-templating", "../../src/aurelia-validation"], function (require, exports, tslib_1, aurelia_dependency_injection_1, aurelia_templating_1, aurelia_validation_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var TriggerForm = /** @class */ (function () {
        function TriggerForm(controller) {
            this.controller = controller;
            this.standardProp = '';
            this.blurProp = '';
            this.focusoutProp = '';
            this.changeProp = '';
            this.changeOrBlurProp = '';
            this.changeOrFocusoutProp = '';
            this.manualProp = '';
            this.focusoutCustomProp = '';
            this.changeOrFocusoutCustomProp = '';
            this.showForm = true;
        }
        Object.defineProperty(TriggerForm.prototype, "focusoutCustomInput", {
            get: function () {
                return this.$focusoutCustomInput || (this.$focusoutCustomInput = this.focusoutCE.querySelector('input'));
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(TriggerForm.prototype, "changeOrFocusoutCustomInput", {
            get: function () {
                return this.$changeOrFocusoutCustomInput
                    || (this.$changeOrFocusoutCustomInput = this.changeOrFocusoutCE.querySelector('input'));
            },
            enumerable: true,
            configurable: true
        });
        TriggerForm = tslib_1.__decorate([
            aurelia_templating_1.inlineView("\n<template>\n  <form novalidate autocomplete=\"off\" if.bind=\"showForm\">\n    <input ref=\"standardInput\"          type=\"text\" value.bind=\"standardProp & validate\">\n    <input ref=\"blurInput\"              type=\"text\" value.bind=\"blurProp & validateOnBlur\">\n    <input ref=\"focusoutInput\"          type=\"text\" value.bind=\"focusoutProp & validateOnFocusout\">\n    <input ref=\"changeInput\"            type=\"text\" value.bind=\"changeProp & validateOnChange\">\n    <input ref=\"changeOrBlurInput\"      type=\"text\" value.bind=\"changeOrBlurProp & validateOnChangeOrBlur\">\n    <input ref=\"changeOrFocusoutInput\"  type=\"text\" value.bind=\"changeOrFocusoutProp & validateOnChangeOrFocusout\">\n    <input ref=\"manualInput\"            type=\"text\" value.bind=\"manualProp & validateManually\">\n    <custom-input ref=\"focusoutCE\"           type=\"text\" value.two-way=\"focusoutCustomProp & validateOnFocusout\"></custom-input>\n    <custom-input ref=\"changeOrFocusoutCE\"   type=\"text\" value.two-way=\"changeOrFocusoutCustomProp & validateOnChangeOrFocusout\"></custom-input>\n    </form>\n</template>"),
            aurelia_dependency_injection_1.inject(aurelia_dependency_injection_1.NewInstance.of(aurelia_validation_1.ValidationController))
        ], TriggerForm);
        return TriggerForm;
    }());
    exports.TriggerForm = TriggerForm;
    aurelia_validation_1.ValidationRules
        .ensure(function (f) { return f.standardProp; }).required()
        .ensure(function (f) { return f.blurProp; }).required()
        .ensure(function (f) { return f.focusoutProp; }).required()
        .ensure(function (f) { return f.focusoutCustomProp; }).required()
        .ensure(function (f) { return f.changeProp; }).required()
        .ensure(function (f) { return f.changeOrBlurProp; }).required().matches(/foo/)
        .ensure(function (f) { return f.changeOrFocusoutProp; }).required().matches(/foo/)
        .ensure(function (f) { return f.changeOrFocusoutCustomProp; }).required().matches(/foo/)
        .ensure(function (f) { return f.manualProp; }).required()
        .on(TriggerForm);
});
