// aurelia-validation/dist/test/test/resources/validation-errors-form-one.js
define(["require", "exports", "tslib", "aurelia-dependency-injection", "aurelia-templating", "../../src/aurelia-validation", "aurelia-framework"], function (require, exports, tslib_1, aurelia_dependency_injection_1, aurelia_templating_1, aurelia_validation_1, aurelia_framework_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ValidationErrorsFormOne = /** @class */ (function () {
        function ValidationErrorsFormOne(controllerFactory) {
            this.controllerFactory = controllerFactory;
            this.standardProp = '';
        }
        ValidationErrorsFormOne.prototype.activate = function (model) {
            this.form = model.form;
            this.model = model;
        };
        ValidationErrorsFormOne.prototype.created = function () {
            var controller = this.model.controller();
            this.controller = controller ? controller : this.controllerFactory.createForCurrentScope();
        };
        ValidationErrorsFormOne.prototype.getViewStrategy = function () {
            return new aurelia_framework_1.InlineViewStrategy(this.form);
        };
        ValidationErrorsFormOne = tslib_1.__decorate([
            aurelia_templating_1.noView,
            aurelia_dependency_injection_1.inject(aurelia_validation_1.ValidationControllerFactory)
        ], ValidationErrorsFormOne);
        return ValidationErrorsFormOne;
    }());
    exports.ValidationErrorsFormOne = ValidationErrorsFormOne;
    aurelia_validation_1.ValidationRules
        .ensure(function (f) { return f.standardProp; }).required()
        .on(ValidationErrorsFormOne);
});
