// aurelia-validation/dist/test/src/config.js
define(["require", "exports", "./validator", "./implementation/standard-validator", "./validate-trigger"], function (require, exports, validator_1, standard_validator_1, validate_trigger_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /**
     * Aurelia Validation Configuration API
     */
    var GlobalValidationConfiguration = /** @class */ (function () {
        function GlobalValidationConfiguration() {
            this.validatorType = standard_validator_1.StandardValidator;
            this.validationTrigger = GlobalValidationConfiguration.DEFAULT_VALIDATION_TRIGGER;
        }
        /**
         * Use a custom Validator implementation.
         */
        GlobalValidationConfiguration.prototype.customValidator = function (type) {
            this.validatorType = type;
            return this;
        };
        GlobalValidationConfiguration.prototype.defaultValidationTrigger = function (trigger) {
            this.validationTrigger = trigger;
            return this;
        };
        GlobalValidationConfiguration.prototype.getDefaultValidationTrigger = function () {
            return this.validationTrigger;
        };
        /**
         * Applies the configuration.
         */
        GlobalValidationConfiguration.prototype.apply = function (container) {
            var validator = container.get(this.validatorType);
            container.registerInstance(validator_1.Validator, validator);
            container.registerInstance(GlobalValidationConfiguration, this);
        };
        GlobalValidationConfiguration.DEFAULT_VALIDATION_TRIGGER = validate_trigger_1.validateTrigger.blur;
        return GlobalValidationConfiguration;
    }());
    exports.GlobalValidationConfiguration = GlobalValidationConfiguration;
});
