// aurelia-validation/dist/test/src/validation-renderer-custom-attribute.js
define(["require", "exports", "tslib", "./validation-controller", "aurelia-templating"], function (require, exports, tslib_1, validation_controller_1, aurelia_templating_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ValidationRendererCustomAttribute = /** @class */ (function () {
        function ValidationRendererCustomAttribute() {
        }
        ValidationRendererCustomAttribute.prototype.created = function (view) {
            this.container = view.container;
        };
        ValidationRendererCustomAttribute.prototype.bind = function () {
            this.controller = this.container.get(validation_controller_1.ValidationController);
            this.renderer = this.container.get(this.value);
            this.controller.addRenderer(this.renderer);
        };
        ValidationRendererCustomAttribute.prototype.unbind = function () {
            this.controller.removeRenderer(this.renderer);
            this.controller = null;
            this.renderer = null;
        };
        ValidationRendererCustomAttribute = tslib_1.__decorate([
            aurelia_templating_1.customAttribute('validation-renderer')
        ], ValidationRendererCustomAttribute);
        return ValidationRendererCustomAttribute;
    }());
    exports.ValidationRendererCustomAttribute = ValidationRendererCustomAttribute;
});
