// aurelia-validation/dist/test/src/validate-trigger.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /**
     * Validation triggers.
     */
    var validateTrigger;
    (function (validateTrigger) {
        /**
         * Manual validation.  Use the controller's `validate()` and  `reset()` methods
         * to validate all bindings.
         */
        validateTrigger[validateTrigger["manual"] = 0] = "manual";
        /**
         * Validate the binding when the binding's target element fires a DOM "blur" event.
         */
        validateTrigger[validateTrigger["blur"] = 1] = "blur";
        /**
         * Validate the binding when it updates the model due to a change in the view.
         */
        validateTrigger[validateTrigger["change"] = 2] = "change";
        /**
         * Validate the binding when the binding's target element fires a DOM "blur" event and
         * when it updates the model due to a change in the view.
         */
        validateTrigger[validateTrigger["changeOrBlur"] = 3] = "changeOrBlur";
        /**
         * Validate the binding when the binding's target element fires a DOM "focusout" event.
         * Unlike "blur", this event bubbles.
         */
        validateTrigger[validateTrigger["focusout"] = 4] = "focusout";
        /**
         * Validate the binding when the binding's target element fires a DOM "focusout" event or
         * when it updates the model due to a change in the view.
         */
        validateTrigger[validateTrigger["changeOrFocusout"] = 6] = "changeOrFocusout";
    })(validateTrigger = exports.validateTrigger || (exports.validateTrigger = {}));
});
